File: basics/frob_norm_mackay.R
frob_norm_mackay=function(n,p,R_hat,th_sig2,X)
{
  th=th_sig2[1]
  sig2=th_sig2[2]
  R=sig2*toeplitz(exp(-th^2*(sin(pi*(0:(p-1))/p))^2)) #for MacKay's periodic kernel
  f=sqrt(sum((R_hat-R)^2))
  return(f)
}

File: MATLAB basics/frob_norm_mackay.m
function frob_norm = frob_norm_mackay(params, n, p, R, X)
    % Computes the Frobenius norm for the optimization
    % Inputs:
    %   params - [theta, sigma^2]
    %   n, p, R, X - Problem parameters
    % Output:
    %   frob_norm - Frobenius norm value
    
    theta = params(1);
    sig2 = params(2);
    
    % Example computation of the Frobenius norm (modify as per exact requirements)
    R_model = sig2 * toeplitz(exp(-theta^2 * (sin(pi * (0:(p-1)) / p)).^2));
    residual = R - R_model;
    frob_norm = norm(residual, 'fro'); % Frobenius norm
end
